starknet_py/net/account/account.py

```diff
@@ -398,9 +402,12 @@ async def sign_declare_v3(
         nonce: Optional[int] = None,
         resource_bounds: Optional[ResourceBoundsMapping] = None,
         auto_estimate: bool = False,
-        tip: int = 0,
+        tip: Optional[int] = None,
     ) -> DeclareV3:
         # pylint: disable=too-many-arguments
+        if tip is None:
+            tip = await get_tips_median(self.client, block_number="latest")
```

Comment:
I'm not sure whether we want to use suggested tip by default. Wouldn't it make more sense if the default value of tip is `0` and someone can pass a flag (e.g. `auto_estimate_tip`) which enforces default estimation? Also, manual tip and mentioned flag would be mutually exclusive. It's similar to how we handle resource bounds, but here the default is 0.